source/main/physics/BeamFactory.cpp

```diff
@@ -164,39 +100,31 @@ ActorManager::ActorManager()
 
     if (RoR::App::app_multithread.GetActive())
     {
-        // Create thread pool
-        int numThreadsInPool = ISETTING("NumThreadsInThreadPool", 0);
+        // Create worker thread (used for physics calculations)
+        m_sim_thread_pool = std::unique_ptr<ThreadPool>(new ThreadPool(1));
 
-        if (numThreadsInPool > 1)
+        // Create general-purpose thread pool
+        if (!BSETTING("DisableThreadPool", false))
         {
-            m_num_cpu_cores = numThreadsInPool;
-        }
-        else
-        {
-            int logical_cpus = hardware_concurrency();
-            int physical_cpus = getNumberOfCPUCores();
+            int logical_cores = std::thread::hardware_concurrency();
+            LOG("BEAMFACTORY: " + TOSTRING(logical_cores) + " logical CPU cores" + " found");
 
-            if (physical_cpus < 6 && logical_cpus > physical_cpus)
-                m_num_cpu_cores = logical_cpus - 1;
+            int numThreadsInPool = ISETTING("NumThreadsInThreadPool", 0);

```

Comment:
I dream of having entire multithreading controlled by single GVarPod_APS<int> `app_threadpool_size` - threadpool of 1 should be equivalent of singlethreaded app - well, mainthread+1background_workerthread in case of RoR, not counting OGRE threads.